crypto/ake/enclave/src/lib.rs

```diff
@@ -196,6 +196,36 @@ impl<EI: EnclaveIdentity> AkeEnclaveState<EI> {
         }
     }
 
+    pub fn frontend_encrypt(
+        &self,
+        session_id: &NonceSession,
+        aad: &[u8],
+        data: &[u8],
+    ) -> Result<EnclaveMessage<NonceSession>> {
+        let mut frontends = self.frontends.lock()?;
+        let session = frontends.get_mut(session_id).ok_or(Error::NotFound)?;
+        let (data, nonce) = session.encrypt_with_nonce(aad, data)?;
+        let channel_id = NonceSession::new(session.binding().to_owned(), nonce);
+
+        // Return message
+        Ok(EnclaveMessage {
+            aad: aad.to_vec(),
+            channel_id,
+            data,
+        })
+    }
+
+    pub fn frontend_decrypt(&self, msg: EnclaveMessage<NonceSession>) -> Result<Vec<u8>> {
+        // Ensure lock gets released as soon as we're done decrypting.
```

Comment:
:wrench: I'm not sure what this comment is trying to communicate.  It feels like it might have come from a scoped block that no longer exists
```suggestion
```

Reply:
removed.